src/main.js

```diff
@@ -145,57 +151,123 @@ let load = base.loader({
     },
   },
 
-  queue: {
-    requires: ['cachemanagers', 'cfg', 'sqs', 'profile', 'monitor'],
-    setup: async ({cachemanagers, cfg, sqs, profile, sqsQueues, monitor}) => {
+  initQueue: {
+    requires: ['cfg', 'sqs'],
+    setup: async ({cfg, sqs}) => {
+      let sqsConfig = cfg.sqsSimple;
+      sqsConfig.sqs = sqs;
+      debug(sqsConfig);
+      return sqsSimple.initQueue(sqsConfig);
+    },
+  },
+
+  queueUrl: {
+    requires: ['cfg', 'sqs'],
+    setup: async ({cfg, sqs}) => {
+      return sqsSimple.getQueueUrl({
+        sqs,
+        queueName: cfg.sqsSimple.queueName,
+      });
+    },
+  },
 
-      let handler = async (obj) => {
-        assert(obj.id, 'must provide id in queue request');
-        assert(typeof obj.id === 'string', 'id must be string');
-        assert(obj.url, 'must provide url in queue request');
-        assert(typeof obj.url === 'string', 'url must be string');
+  deadQueueUrl: {
+    requires: ['cfg', 'sqs', 'profile'],
+    setup: async ({cfg, sqs, profile}) => {
+      return sqsSimple.getQueueUrl({
+        sqs,
+        queueName: cfg.sqsSimple.queueName + cfg.sqsSimple.deadLetterSuffix,
+      });
+    },
+  },
 
-        let selectedCacheManagers = cachemanagers.filter(x => x.id === obj.id);
-        await Promise.all(selectedCacheManagers.map(x => x.put(obj.url)));
-      };
+  queueSender: {
+    requires: ['cfg', 'sqs', 'queueUrl'],
+    setup: async ({cfg, sqs, queueUrl}) => {
+      return new sqsSimple.QueueSender({sqs, queueUrl});
+    },
+  },
+
+  queueListenerFactory: {
+    requires: ['cfg', 'sqs', 'queueUrl', 'cacheManagers', 'profile', 'monitor'],
+    setup: async ({cfg, sqs, queueUrl, cacheManagers, profile, monitor}) => {
+      return async function() {
```

Comment:
because I am listening on 1-N queues, and I want to be able to generate more than one from main.js, but that requires returning a factory since calling the loader twice returns the same object.